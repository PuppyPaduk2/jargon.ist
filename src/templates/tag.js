// @flow

import React from 'react';
import Helmet from 'react-helmet';
import Link from 'gatsby-link';
import { graphql } from 'graphql';

import { Container, Section } from '../components/ui';

type Props = {
  data: Object,
  pathContext: Object,
};

const Jargon = ({ data, pathContext }: Props) => {
  const { allMarkdownRemark: { totalCount, edges: jargons } } = data;

  return (
    <Section>
      <Container>
        <Helmet>
          <title>{`${pathContext.tag} konulu jargonlar`}</title>
        </Helmet>
        <h1>
          {pathContext.tag} ({totalCount})
        </h1>
        <ul style={{ margin: 0, padding: 0, listStyle: 'none' }}>
          {jargons.map(jargon => (
            <li key={jargon.node.fields.slug} className="u-gap-bottom-xsmall">
              <Link to={`/${jargon.node.fields.slug}`} className="u-text-lowercase">
                {jargon.node.frontmatter.title}
              </Link>
            </li>
          ))}
        </ul>
      </Container>
    </Section>
  );
};

export const pageQuery = graphql`
  query JargonsByTag($tag: String!) {
    allMarkdownRemark(limit: 1000, filter: { frontmatter: { tags: { in: [$tag] } } }) {
      totalCount
      edges {
        node {
          fields {
            slug
          }
          frontmatter {
            title
          }
        }
      }
    }
  }
`;

export default Jargon;
