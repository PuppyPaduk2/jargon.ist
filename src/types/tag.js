// @flow

export type Tag = {
  title: string,
  slug: string,
};
