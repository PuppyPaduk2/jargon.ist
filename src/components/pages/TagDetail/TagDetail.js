// @flow

import React, { Fragment } from 'react';
import Link from 'gatsby-link';

import { type Jargon } from '../../../types/jargon';

type Props = {
  tag: string,
  jargons: Array<Jargon>,
  totalCount: number,
};

const TagDetail = ({ tag, jargons, totalCount }: Props) => (
  <Fragment>
    <h1>
      {tag} ({totalCount})
    </h1>
    <ul style={{ margin: 0, padding: 0, listStyle: 'none' }}>
      {jargons.map(jargon => (
        <li key={jargon.slug || ''} className="u-gap-bottom-xsmall">
          <Link to={`/${jargon.slug || ''}`} className="u-text-lowercase">
            {jargon.title}
          </Link>
        </li>
      ))}
    </ul>
  </Fragment>
);

export default TagDetail;
