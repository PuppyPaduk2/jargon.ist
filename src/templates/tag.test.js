// @flow

import React from 'react';
import { shallow } from 'enzyme';

import { Tag } from './tag';

import TagDetail from '../components/pages/TagDetail';

const data = {
  allMarkdownRemark: {
    totalCount: 0,
    edges: [
      {
        node: {
          fields: {
            slug: 'hello',
          },
          frontmatter: { title: 'hello' },
        },
      },
    ],
  },
};

it('renders correctly', () => {
  const wrapper = shallow(<Tag data={data} pathContext={{ tag: 'tag1' }} />);

  expect(wrapper).toMatchSnapshot();
});

it('renders title', () => {
  const wrapper = shallow(<Tag data={data} pathContext={{ tag: 'tag1' }} />);

  expect(wrapper.find('title')).toHaveLength(1);
});

it('renders TagDetail component', () => {
  const wrapper = shallow(<Tag data={data} pathContext={{ tag: 'tag1' }} />);

  expect(wrapper.find(TagDetail)).toHaveLength(1);
});
