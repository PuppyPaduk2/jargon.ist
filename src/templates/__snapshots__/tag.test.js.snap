// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<React.Fragment>
  <HelmetWrapper
    defer={true}
    encodeSpecialCharacters={true}
  >
    <title>
      tag1 konulu jargonlar
    </title>
    <meta
      content="tag1 konulu jargonlar"
      name="og:title"
    />
    <meta
      content="tag1 konulu jargonlar"
      name="twitter:title"
    />
  </HelmetWrapper>
  <TagDetail
    jargons={
      Array [
        Object {
          "html": "",
          "slug": "hello",
          "title": "hello",
        },
      ]
    }
    tag="tag1"
    totalCount={0}
  />
</React.Fragment>
`;
